src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -170,137 +118,112 @@ internal void DependencyRegen(WorkspaceModel ws)
+        /// <param name="e"></param>
+        private void DownloadPackage(object sender, RoutedEventArgs e)
         {
-            get; set;
+            var info = ((NodeLibraryDependencyRow)((Button)sender).DataContext).DependencyInfo;
+            var package = new PackageInfo(info.Name, info.Version);
+
+            packageInstaller.DownloadAndInstallPackage(package);
+            DependencyRegen(currentWorkspace);
         }
     }
 
-    /// <summary>
-    /// Class defining data for each column
-    /// </summary>
-    public class ColumnData
+    public class NodeLibraryDependencyRow
     {
-        static int DefaultWidth = 200;
-        static SolidColorBrush DefaultColor = (SolidColorBrush)(new BrushConverter().ConvertFrom("#aaaaaa"));
-        public static Brush MissingColor = Brushes.Red;
-        public static Brush WarningColor = Brushes.Yellow;
+        public INodeLibraryDependencyInfo DependencyInfo { get; private set; }
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        public ColumnData(string data)
+        public NodeLibraryDependencyRow(INodeLibraryDependencyInfo nodeLibraryDependencyInfo)
         {
-            Data = data;
-            Width = DefaultWidth;
-            Color = DefaultColor;
+            DependencyInfo = nodeLibraryDependencyInfo;
         }
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        /// <param name="width"></param>
-        public ColumnData(string data, int width)
-        {
-            Data = data;
-            Width = width;
-            Color = DefaultColor;
-        }
+        public string Name => DependencyInfo.Name;
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        /// <param name="color"></param>
-        public ColumnData(string data, Brush color)
-        {
-            Data = data;
-            Width = DefaultWidth;
-            Color = color;
-        }
+        public Version Version => DependencyInfo.Version;
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        /// <param name="width"></param>
-        /// <param name="brush"></param>
-        public ColumnData(string data, int width, Brush brush)
+        public string DetailsMessage
         {
-            Data = data;
-            Width = width;
-            Color = brush;
-        }
+            get
+            {
+                string message;
 
-        /// <summary>
-        /// Data in each cell
-        /// </summary>
-        public string Data
-        {
-            get; set;
-        }
+                switch (DependencyInfo.State)
+                {
+                    case PackageDependencyState.Loaded:
+                        message = string.Format("{0} {1} is installed locally.", 
```

Comment:
@mjkkirschner @QilongTang While this view extension is in "Preview", can we leave user facing strings in the code rather than move them to resources? I imagine these messages will be changes as we add other features such as "Keep Local"

Reply:
thats okay with me as long as tasks are filed and TODOs added referencing them.

Reply:
Ok, sounds like less work to just go ahead and move them to resources, so I'll do that

Reply:
Moved to resx